packages/@ember/-internals/glimmer/lib/component-managers/curly.ts

```diff
@@ -133,6 +134,12 @@ export default class CurlyComponentManager
     let factory: TemplateFactory;
 
     if (layout === undefined) {
+      let theSetTemplate = getComponentTemplate(component);
```

Comment:
a chance to early exit in classic land without resolving a layout

Reply:
This is a little surprising in the context of a deprecation PR. Do we have a bug here? Like, are there cases where an `@ember/component` doesn't support template colocation without this change?

If there's a bugfix we might want to land it separately from adding the deprecation. And make sure it has its own test coverage. (Not sure if this PR includes such coverage, haven't found it yet.)

Reply:
>  doesn't support template colocation without this change?

it does -- and I confirmed by adding a specific test about `getComponentTemplate` and removing the code here -- however -- this code is about looking up the template for the debug tree -- which is no longer a problem -- so I'm going to just remove this code as it looks like all the tests pass.

There isn't a bug to fix, imo.